synapse_bind_sydent/__init__.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Any, Dict
+from urllib.parse import urlparse
+
+import attr
+from synapse.module_api import ModuleApi
+from synapse.module_api.errors import ConfigError
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True, frozen=True)
+class SydentBinderConfig:
+    sydent_base_url: str
+
+
+class SydentBinder:
+    def __init__(self, config: SydentBinderConfig, api: ModuleApi) -> None:
+        # Keep a reference to the config and Module API
+        self._api = api
+        self._config = config
+
+        self._http_client = api.http_client
+
+        self._sydent_bind_url = (
+            f"{config.sydent_base_url}/_matrix/identity/internal/bind"
+        )
+
+        scheme = urlparse(config.sydent_base_url).scheme
+        self._sydent_host = config.sydent_base_url.replace(f"{scheme}://", "")
```

Comment:
Curious why you'd go so far to parse the URL and then chuck that away and fall back to string replacement? :sweat_smile: 

Do you want the hostname (`https://sydent/` → `sydent`) or the hostname with port (`https://sydent:4242/` → `sydent:4242`) (this also includes username and password if specified, but meh...)?

```suggestion
        self._sydent_host = urlparse(config.sydent_base_url).hostname
```

or

```suggestion
        self._sydent_host = urlparse(config.sydent_base_url).netloc
```

Reply:
Reading 
```
            id_server: The identity server the 3PID association has been registered on.
                This should only be the domain (or IP address, optionally with the port
                number) for the identity server. This will be used to reach out to the
                identity server using HTTPS (unless specified otherwise by Synapse's
                configuration) when attempting to unbind the third-party identifier.
```

I think you really don't want any potential path component to creep in. I think you want `netloc` in this case.